File: fibonacci_series/implementation/java/Fibonacci.java
import java.util.*;
import java.math.BigInteger;

public class Fibonacci{
    public static void main(String args[]){
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the postion of fibonnaci number (n) to calculate : ");
        int n = input.nextInt();
        if(n < 0){
            System.out.println("n can't be less than 0");
        }else if(n <= 2){
            System.out.println(n - 1);
        }else{
            // We'll use BigIntegers to store our values as fibonacci numbers can get really long
            BigInteger secondLast = new BigInteger("0");
            BigInteger last = new BigInteger("1");
            BigInteger fibonnaciNumber = new BigInteger(secondLast.toString());
            fibonnaciNumber = fibonnaciNumber.add(last);
            for(int i = 3; i < n; i++){
                // Since BigIntegers are immutable we need to reallocate them instead of reassign
                secondLast = new BigInteger(last.toString());
                last = new BigInteger(fibonnaciNumber.toString());
                fibonnaciNumber = new BigInteger(secondLast.toString());
                fibonnaciNumber = fibonnaciNumber.add(last);
            }
            System.out.println(fibonnaciNumber);
        }
    }
}

File: fibonacci_series/implementation/python/fibonacci.py
def fibonacci_value(n):
    if n < 0:
        print("n can't be less than 0")
    elif n <= 2:
        print(n - 1)
    else:
        second_last = 0
        last = 1
        fibonacci_num = second_last + last
        for _ in range(3, n):
            second_last = last
            last = fibonacci_num
            fibonacci_num = second_last + last
        return fibonacci_num

if __name__ == "__main__":
    n = int(input("Enter the postion of fibonnaci number (n) to calculate : "))
    print(fibonacci_value(n))
